File: src/main/java/pro/sdacademy/zdjava137/group3/entity/Category.java
package pro.sdacademy.zdjava137.group3.entity;

import jakarta.persistence.*;
import jakarta.validation.constraints.NotEmpty;
import jakarta.validation.constraints.Size;
import lombok.Data;

import java.util.List;

@Data
@Entity
public class Category {
    @Id
    @GeneratedValue
    private long id;
    @NotEmpty
    @Size(min = 3)
    private String name;
    private String description;
    @ManyToOne
    private Category parent;
    @OneToMany(mappedBy = "parent")
    private List<Category> children;
}

File: frontend/src/app/service/model/category.ts
export interface Category {
  id: number
  name: string
  description: string
  parent: Category | undefined | null
}

export type Categories = Category[]

export interface CategoryCreate {
  name: string
  description: string
  parent_id: number | undefined
}
export interface CategoryUpdate {
  name: string | undefined
  description: string | undefined
  parent_id: number | undefined
}
